Raven.Database/FileSystem/Storage/Esent/Schema/Updates/From08To09.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.Isam.Esent.Interop;
+using Raven.Database.Config;
+
+namespace Raven.Database.FileSystem.Storage.Esent.Schema.Updates
+{
+    public class From08To09 : IFileSystemSchemaUpdate
+    {
+        public string FromSchemaVersion
+        {
+            get { return "0.8"; }
+        }
+        public void Init(InMemoryRavenConfiguration configuration)
+        {
+        }
+
+        public void Update(Session session, JET_DBID dbid, Action<string> output)
+        {
+            var pagesTableName = "pages";
+
+            var newTableName = pagesTableName + "_new";
+
+            try
+            {
+                Api.JetDeleteTable(session, dbid, newTableName);
+            }
+            catch (Exception)
+            {
+                //if there is no such table - then it is not important
+                //this is a precaution against partially failed upgrade process
+            }
+
+            SchemaCreator.CreatePagesTable(dbid, newTableName, session);
+
+            var buffer = new byte[StorageConstants.MaxPageSize];
+            var bookMarkBuffer = new byte[SystemParameters.BookmarkMost];
+
+            using (var usage = new Table(session, dbid, "usage", OpenTableGrbit.None))
+            {
+                // let's create index in usages table necessary for migration purposes
+                // we need to seek by page_id in order to update value in 'usages' table after
+                // writing a page to new table where it will get new id
+
+                var indexDef = "+page_id\0\0";
+                try
+                {
+                    Api.JetDeleteIndex(session, usage, "by_page_id");
+                }
+                catch (Exception)
+                {
+                    //if there is no such index - then it is not important
+                    //this is a precaution against partially failed upgrade process
+                }
+
+                Api.JetCreateIndex2(session, usage, new[]
+                {
+                    new JET_INDEXCREATE
+                    {
+                        szIndexName = "by_page_id",
+                        cbKey = indexDef.Length,
+                        cbKeyMost = SystemParameters.KeyMost,
+                        cbVarSegMac = SystemParameters.KeyMost,
+                        szKey = indexDef,
+                        grbit = CreateIndexGrbit.None,
+                        ulDensity = 80
+                    }
+                }, 1);
+
+                // commit index creation
```

Comment:
This can take some time, maybe report so user have visibility?

Reply:
I've changed that in https://github.com/ravendb/ravendb/pull/7205/commits/a4a2d246a22bda6991882f7f3a6740e93ef217a4

We'll create that index in "new" 'usage' table at the time of From07To08 migration and feed it then. Next we'll use it during From08To09 and delete it once we're done